iOSDeviceManager/Utilities/XCTestConfigurationPlist.m

```diff
@@ -4,468 +4,320 @@
 
 @implementation XCTestConfigurationPlist
 
-+ (NSString *)plistWithTestBundlePath:(NSString *)testBundlePath {
-    NSString *plist = nil;
++ (NSString *)plistWithXCTestInstallPath:(NSString *)testInstallPath
+                        AUTInstalledPath:(NSString *)autInstalledPath
+                     AUTBundleIdentifier:(NSString *)autBundleIdentifier
+                     runnerInstalledPath:(NSString *)runnerInstalledPath
+                  runnerBundleIdentifier:(NSString *)runnerBundleIdentifier
+                       sessionIdentifier:(NSString *)UUID{
+    NSString *plist = [[XCTestConfigurationPlist plist] mutableCopy];
```

Comment:
As written, you do not need to make a mutable copy - The `.plist` method will return a non-constant string.

`mutableCopy` returns an `NSMutableString`.

If you want to use `NSMutableString`, then use this method NSMutableString:

```
- (NSUInteger)replaceOccurrencesOfString:(NSString *)target 
                              withString:(NSString *)replacement 
                                 options:(NSStringCompareOptions)options 
                                   range:(NSRange)searchRange;
```

Reply:
Put another way, if you want to keep the code as-is, just don't make a mutable copy.

If you want to use NSMutableString, use the method I pasted.